storage/internal/elastic/queries.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2019 Bull S.A.S. Atos Technologies - Bull, Rue Jean Jaures, B.P.68, 78340, Les Clayes-sous-Bois, France.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elastic
+
```

Comment:
For this part,  maybe you can use a structure you marshall in JSON ?

Reply:
I though about that, should be more readable and more extensible and maintainable. Just didn't want to add masrhaling operation ... but maybe I'm too maniaco ... Let's plan to do this !